bioptim/optimization/optimal_control_program.py

```diff
@@ -406,6 +416,7 @@ def __init__(
             # Inner- and inter-phase continuity
             ContinuityFunctions.continuity(self)
 
+        # @TODO: ajouter un check
```

Comment:
I think the check you refer too is what is done in lines 386-392. No need for more